Configurations/15-android.conf

```diff
@@ -0,0 +1,207 @@
+# you need to target older platform, pass additional -D__ANDROID_API__=N
+# to Configure. For example, to compile for ICS on ARM with NDK 10d:
+#
+# ANDROID_NDK=/some/where/android-ndk-10d
+# PATH=$ANDROID_NDK/toolchains/arm-linux-androideabi-4.8/prebuild/linux-x86_64/bin:$PATH
+# [..]./Configure android-arm -D__ANDROID_API__=14
+#
+# One can engage clang by passing CC=clang to Configure. In such case
+# PATH needs even more adjustments to cover NDK's clang itself, as well
+# as unprefixed, yet target-specific ar and ranlib [or not, if you use
+# binutils-multiarch].
+
+{
+    my $android_ndk = {};
+    my %triplet = (
+        arm    => "arm-linux-androideabi",
+        arm64  => "aarch64-linux-android",
+        mips   => "mipsel-linux-android",
+        mips64 => "mips64el-linux-android",
+        x86    => "i686-linux-android",
+        x86_64 => "x86_64-linux-android",
+    );
+
+    sub android_ndk {
+        unless (%$android_ndk) {
+            my $ndk = $ENV{ANDROID_NDK};
+            die "\$ANDROID_NDK is not defined"  if (!$ndk);
+            die "\$ANDROID_NDK=$ndk is invalid" if (!-d "$ndk/platforms");
+
+            my $sysroot;
+
+            if (!($sysroot = $ENV{CROSS_SYSROOT})) {
+                my $api = "*";
+
+                # see if user passed -D__ANDROID_API__=N
+                foreach (@{$useradd{CPPDEFINES}}) {
+                    if (m|__ANDROID_API__=([0-9]+)|) {
+                        $api = $1;
+                        last;
+                    }
+                }
+
+                # list available platforms [numerically]
+                my @platforms = sort { $a =~ m/-([0-9]+)$/; my $aa = $1;
+                                       $b =~ m/-([0-9]+)$/; $aa <=> $1;
+                                     } glob("$ndk/platforms/android-$api");
+                die "no $ndk/platforms/android-$api" if ($#platforms < 0);
+
+                $config{target} =~ m|[^-]+-([^-]+)$|;   # split on dash
+                $sysroot = "@platforms[$#platforms]/arch-$1";
+            }
+            die "no sysroot=$sysroot"   if (!-d $sysroot);
+
+            $sysroot =~ m|/android-([0-9]+)/arch-(\w+)/?$|;
+            my ($api, $arch) = ($1, $2);
+
+            my $triarch = $triplet{$arch};
+            my $cflags = "-Wa,--noexecstack";
+            my $cppflags;
+
+            # see if user passed CC=clang
+            if ($user{CC} eq "clang") {
+                if (which("clang") !~ m|^$ndk/.*/prebuilt/([^/]+)/|) {
+                    die "no clang in \$ANDROID_NDK";
+                }
+                $cflags .= " -target $triarch -gcc-toolchain "
+                        . "\$(ANDROID_NDK)/toolchains/$triarch-4.9/prebuilt/$1";
+                $user{CROSS_COMPILE} = undef;
+            } else {
+                $cflags .= " -mandroid";
+                $user{CROSS_COMPILE} = "$triarch-";
+            }
+
+            if (!-d "$sysroot/usr/include") {
```

Comment:
ndk r14* and r15* are with "conventional" headers located in sysroot and unified headers. Unfortunately both versions are not compatible. I prefer code to use "conventional" headers in such case.

Reply:
And this line check if there are headers in sysroot and uses them, i.e. it does default to "conventional" headers.